app.rb

```diff
@@ -2,93 +2,101 @@
 require './student'
 require './teacher'
 require './classroom'
-require './rental'
 require './book'
 
 class App
-  def initialize()
-    @all_books = []
-    @all_persons = []
+  def initialize(person_manager, book_manager, rental_manager)
+    @person_manager = person_manager
+    @book_manager = book_manager
+    @rental_manager = rental_manager
   end
 
-  def create_book(title, author)
-    book = Book.new(title, author)
-    @all_books.push(book)
-    puts 'Book created successfully'
-  end
-
-  def create_person(person_code)
-    if person_code == 1
-      create_student
-    elsif person_code == 2
-      create_teacher
+  def execute_option(number)
+    case number
+    when 1
+      @book_manager.list_books
+    when 2
+      @person_manager.list_persons
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental(@book_manager.all_books, @person_manager.all_persons)
+    when 6
+      get_rental(@person_manager.all_persons)
     else
       puts 'Enter valid number'
-      call(3)
+      main
     end
-    puts 'Person created successfully'
   end
 
-  def create_student
-    print 'Age: '
-    age = gets.chomp.to_i
-    print 'Name: '
-    name = gets.chomp
-    print 'Has parent permission? [Y/N]: '
-    permission = gets.chomp
-    permission_values = %w[n N]
-    person = Student.new(age, name, permission_values.include?(permission))
-    @all_persons.push(person)
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @book_manager.create_book(title, author)
+    puts 'Book created successfully'
   end
 
-  def create_teacher
+  def create_person
+    print 'Do you want to create student (1) or a teacher (2)? [Input number]: '
+    person_code = gets.chomp.to_i
     print 'Age: '
     age = gets.chomp.to_i
     print 'Name: '
     name = gets.chomp
-    print 'Specialization: '
-    specialization = gets.chomp
-    person = Teacher.new(age, specialization, name)
-    @all_persons.push(person)
-  end
-
-  def list_books
-    puts @all_books.inspect
-    @all_books.each do |book|
-      puts "Title: \"#{book.title}\", Author: #{book.author}"
-    end
-  end
-
-  def list_persons
-    @all_persons.each do |person|
-      person_type = person.instance_of?(Student) ? 'Student' : 'Teacher'
-      puts "[#{person_type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
-    end
+    check_person_code(person_code, age, name)
+    puts 'Person created successfully'
```

Comment:
- [ ] I really appreciate the manager classes you've created separating the concerns, but you've missed one small requirement according to the single responsibility principle It's best if all requests for input and input validation are placed in separate classes/methods whose responsibilities are to request or validate input. Refer to this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for further guidance.
- [ ] This comment applies to other methods where you're following similar patterns so please carefully investigate and refactor your code accordingly.